tests/datasets/test_seasonet.py

```diff
@@ -0,0 +1,210 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+import glob
+import os
+import shutil
+from pathlib import Path
+from typing import Union
+
+import matplotlib.pyplot as plt
+import pytest
+import requests
+import torch
+import torch.nn as nn
+from _pytest.fixtures import SubRequest
+from _pytest.monkeypatch import MonkeyPatch
+from torch.utils.data import ConcatDataset
+
+import torchgeo.datasets.utils
+from torchgeo.datasets import SeasoNet
+
+
+class MockResponse:
```

Comment:
Why is this class necessary? We haven't used anything like this in other datasets

Reply:
See other comments. Handled in one of the commits.